x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -662,21 +663,75 @@ func setPaths() error {
 	if err := syncDir(srcDownloads, destDownloads); err != nil {
 		return fmt.Errorf("syncing download directory to STATE_PATH(%s) failed: %s", statePath, err)
 	}
+	originalTop := paths.Top()
 	paths.SetTop(topPath)
 	paths.SetConfig(configPath)
 	// when custom top path is provided the home directory is not versioned
 	paths.SetVersionHome(false)
 	// set LOGS_PATH is given
-	if logsPath := envWithDefault("", "LOGS_PATH"); logsPath != "" {
+	logsPath = envWithDefault(logsPath, "LOGS_PATH")
+	if logsPath != "" {
 		paths.SetLogs(logsPath)
 		// ensure that the logs directory exists
 		if err := os.MkdirAll(filepath.Join(logsPath), 0755); err != nil {
 			return fmt.Errorf("preparing LOGS_PATH(%s) failed: %s", logsPath, err)
 		}
 	}
+	// persist the paths so other commands in the container will use the correct paths
+	if writePaths {
+		if err := writeContainerPaths(originalTop, statePath, configPath, logsPath); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type containerPaths struct {
+	StatePath  string `config:"state_path" yaml:"state_path"`
+	ConfigPath string `config:"state_path" yaml:"config_path,omitempty"`
+	LogsPath   string `config:"state_path" yaml:"logs_path,omitempty"`
+}
+
+func writeContainerPaths(original, statePath, configPath, logsPath string) error {
+	pathFile := filepath.Join(original, "container-paths.yml")
```

Comment:
Do we need a new file for this or could we use an existing "state" file?

Reply:
It is only specific to running in container mode. Being that Elastic Agent ships with pre-built binaries and the state will normally be mounted from a volume we need to do a lot to ensure that the containers paths are correct.

The code will not know the path of `state.yml` because that is based on the `STATE_PATH` adjusting that location. So we use a unique file to just do this one thing, for the `status` and `inspect` commands. All other commands should either not be used in the container or are used as children of the `container` command itself.